src/client/mod.rs

```diff
@@ -75,6 +77,11 @@ impl<C: NetworkConnector> Client<C> {
         self.redirect_policy = policy;
     }
 
+    /// Set the UserAgent. Pass `None` to disable.
+    pub fn set_user_agent(&mut self, agent: Option<String>) {
```

Comment:
What if instead of a specific method for this header, we just have a set of headers we clone into every request, so users can set any default headers really easily and the API is really simple.